cmd/batch-expire.go

```diff
@@ -653,6 +670,11 @@ func (r *BatchJobExpire) Start(ctx context.Context, api ObjectLayer, job BatchJo
 			ObjectInfo: result.Item,
 		})
 	}
+	if context.Cause(ctx) != nil {
+		xioutil.SafeClose(expireCh)
+		xioutil.SafeClose(saverQuitCh)
```

Comment:
If the context is canceled the saver will exit anyway.

Reply:
`context.Cause(ctx) != nil` implies that ctx is canceled... Then observe the code:

```
			case <-ctx.Done():
				// persist in-memory state immediately before exiting due to context cancellation.
				batchLogIf(ctx, ri.updateAfter(ctx, api, 0, job))
				return
			case <-saverQuitCh:
				// persist in-memory state immediately to disk.
				batchLogIf(ctx, ri.updateAfter(ctx, api, 0, job))
				return
```